kubemarine/system.py

```diff
@@ -48,30 +48,36 @@ def verify_inventory(inventory, cluster):
 
 
 def enrich_etc_hosts(inventory, cluster):
+# enrich only etc_hosts_generated object, etc_hosts remains as it is
+
+    # if by chance cluster.yaml contains non empty etc_hosts_generated we have to reset it
+    inventory['services']['etc_hosts_generated'] = {}
+
     control_plain = inventory['control_plain']['internal']
 
+    # take custom name of control_plain from etc_hosts, if any
     control_plain_names = inventory['services']['etc_hosts'].get(control_plain, [])
     control_plain_names.append(cluster.inventory['cluster_name'])
     control_plain_names.append('control-plain')
     control_plain_names = list(OrderedSet(control_plain_names))
-    inventory['services']['etc_hosts'][control_plain] = control_plain_names
+    inventory['services']['etc_hosts_generated'][control_plain] = control_plain_names
 
     for node in cluster.inventory['nodes']:
         if 'remove_node' in node['roles']:
             continue
 
-        internal_node_ip_names = inventory['services']['etc_hosts'].get(node['internal_address'], [])
+        internal_node_ip_names = inventory['services']['etc_hosts_generated'].get(node['internal_address'], [])
         internal_node_ip_names.append("%s.%s" % (node['name'], cluster.inventory['cluster_name']))
         internal_node_ip_names.append(node['name'])
         internal_node_ip_names = list(OrderedSet(internal_node_ip_names))
-        inventory['services']['etc_hosts'][node['internal_address']] = internal_node_ip_names
+        inventory['services']['etc_hosts_generated'][node['internal_address']] = internal_node_ip_names
```

Comment:
This code becomes too complex. It was designed to add new names to list of custom names in `etc_hosts`, but we do not support custom names in `etc_hosts_generated`.

Reply:
reworked.